### PR TITLE
Flan T5: Read lm_head when word embeddings are not tied

### PR DESCRIPTION
**Pull request by jbochi**
In https://github.com/huggingface/candle/pull/840 we added support for Flan T5 and checked the output of the encoder was correct. However, when I added decoding in https://github.com/huggingface/candle/pull/864, I didn't check it also worked for FlanT5. This fixes it.

```bash
$ cargo run --release --example t5 -- --model-id "google/flan-t5-base" --prompt "summarize: state authorities dispatched emergency crews tuesday to survey the damage after an onslaught of severe weather in mississipi." --temperature 0 --decode
   Compiling candle-transformers v0.2.3 (/Users/jbochi/dev/github/huggingface/candle/candle-transformers)
   Compiling candle-examples v0.2.3 (/Users/jbochi/dev/github/huggingface/candle/candle-examples)
    Finished release [optimized] target(s) in 6.04s
     Running `target/release/examples/t5 --model-id google/flan-t5-base --prompt 'summarize: state authorities dispatched emergency crews tuesday to survey the damage after an onslaught of severe weather in mississipi.' --temperature 0 --decode`
Running on CPU, to run on GPU, build this example with `--features cuda`
 weather in mississippi slams state
15 tokens generated (13.95 token/s)
```

I also confirmed the lm head weights and the logits match the pytorch implementation.

### CODE REVIEW

**Review by LaurentMazare — APPROVED** [review]

**Comment by LaurentMazare** [comment]
Could you run rustfmt please?

**Comment by jbochi** [comment]
> Could you run rustfmt please?

Ooops. Done. I need to fix my editor settings.

PS: I also tested this change with a model for grammar correction from Grammarly, my employer:

```
 $ cargo run --release --example t5 -- --model-id "grammarly/coedit-large" --revision "refs/pr/4"   --prompt "Fix the grammar: When I grow up, I start to understand what he said is quite right." --temperature 0 --decode
   Compiling candle-transformers v0.2.3 (/Users/jbochi/dev/github/huggingface/candle/candle-transformers)
   Compiling candle-examples v0.2.3 (/Users/jbochi/dev/github/huggingface/candle/candle-examples)
    Finished release [optimized] target(s) in 6.58s
     Running `target/release/examples/t5 --model-id grammarly/coedit-large --revision refs/pr/4 --prompt 'Fix the grammar: When I grow up, I start to understand what he said is quite right.' --temperature 0 --decode`
Running on CPU, to run on GPU, build this example with `--features cuda`
 When I grow up, I will start to understand what he said is quite right.
19 tokens generated (4.72 token/s)
```

File: candle-transformers/src/models/t5.rs [hunk 1]
```diff
@@ -653,12 +677,19 @@ impl T5ForConditionalGeneration {
         let decoder_output = self
             .decoder
             .forward(decoder_input_ids, Some(encoder_output))?;
-        let sequence_output = decoder_output
+        let mut sequence_output = decoder_output
```
Comment (LaurentMazare) [inline candle-transformers/src/models/t5.rs:680]: Avoid using `mut` here as it's easy to do a `let ... = if self.tie_word_embeddings {} else {};` instead.

File: candle-transformers/src/models/t5.rs [hunk 2]
```diff
@@ -633,9 +644,22 @@ impl T5ForConditionalGeneration {
         decoder_cfg.num_layers = cfg.num_decoder_layers.unwrap_or(cfg.num_layers);
         let decoder = T5Stack::load(true, vb.pp("decoder"), &shared, &decoder_cfg)?;
 
+        let tie_word_embeddings = cfg.tie_word_embeddings;
+        let lm_head = match tie_word_embeddings {
```
Comment (LaurentMazare) [inline candle-transformers/src/models/t5.rs:648]: Use a `if` when matching on a `bool` if possible (it's what we do everywhere else in this codebase).

File: candle-transformers/src/models/t5.rs (hunk 1 above)
Comment (jbochi) [reply candle-transformers/src/models/t5.rs]: Done. PTAL.

File: candle-transformers/src/models/t5.rs (hunk 2 above)
Comment (jbochi) [reply candle-transformers/src/models/t5.rs]: Done. PTAL.

**Review by LaurentMazare — APPROVED** [review]

**Comment by LaurentMazare** [comment]
Thanks!
